lib/ASTGen/Sources/ASTGen/Types.swift

```diff
@@ -33,13 +33,14 @@ func isTypeMigrated(_ node: TypeSyntax) -> Bool {
   while true {
     switch current.kind {
     case // Known implemented kinds.
-        .arrayType, .attributedType, .compositionType, .someOrAnyType,
-        .dictionaryType, .functionType, .implicitlyUnwrappedOptionalType,
-        .memberType, .metatypeType, .namedOpaqueReturnType, .optionalType,
-        .packExpansionType, .identifierType, .tupleType:
+        .arrayType, .attributedType, .classRestrictionType, .compositionType,
+        .someOrAnyType, .dictionaryType, .functionType, .identifierType,
+        .implicitlyUnwrappedOptionalType, .memberType, .metatypeType,
+        .namedOpaqueReturnType, .optionalType, .packElementType,
+        .packExpansionType, .suppressedType, .tupleType:
       break
     case // Known unimplemented kinds.
-        .suppressedType, .packElementType, .missingType:
+        .missingType:
       return false;
```

Comment:
Are all the types migrated now? MissingTypeSyntax should only occur in invalid syntax trees AFAIK, I guess we just need some diagnostic logic for it?

Reply:
For valid syntax, yeah I think so. As for the diagnostics, I _think_ the parser already emitted some. I feel like just generating `ErrorTypeRepr` might be enough, but I'm not sure. Maybe we need to emit `nullptr` for the entire type syntax.